WordPress/src/androidTest/java/org/wordpress/android/e2e/pages/EditorPage.java

```diff
@@ -5,52 +5,99 @@
-
-    private static ViewInteraction titleField = onView(allOf(withId(R.id.title), withHint("Title")));
-
-    private static ViewInteraction publishConfirmation = onView(allOf(
-            withText("Post published"), isDescendantOfA(withId(R.id.snackbar))));
+    private static ViewInteraction titleField = onView(allOf(withId(R.id.title),
+            withHint("Title")));
+    private static ViewInteraction publishConfirmation = onView(withText("Post published"));
+    private static ViewInteraction addMediaButton = onView(withId(R.id.media_button_container));
+    private static ViewInteraction mediaLibraryButton = onView(withId(R.id.media_bar_button_library));
+    private static ViewInteraction allowMediaAccessButton = onView(allOf(withId(R.id.button),
+            withText("Allow")));
+    private static ViewInteraction confirmButton = onView(withId(R.id.mnu_confirm_selection));
 
     public EditorPage() {
         editor.check(matches(isDisplayed()));
     }
 
-    public EditorPage enterTitle(String postTitle) {
+    public void enterTitle(String postTitle) {
         titleField.perform(typeText(postTitle), ViewActions.closeSoftKeyboard());
-
-        return this;
     }
 
-    public EditorPage enterContent(String postContent) {
+    public void enterContent(String postContent) {
         editor.perform(typeText(postContent), ViewActions.closeSoftKeyboard());
+    }
+
+    // Image needs a little time to be uploaded after entering the image
+    public void enterImage() {
+        // Click on add media button
+        clickOn(addMediaButton);
+        clickOn(mediaLibraryButton);
+
+        if (isElementDisplayed(allowMediaAccessButton)) {
+            // Click on Allow button
+            clickOn(allowMediaAccessButton);
+        }
+
+        // Click on a random image
+        onView(withIndex(withId(R.id.media_grid_item_image), 0)).perform(click());
+
+        // Click the confirm button
+        clickOn(confirmButton);
+
+        if (isElementDisplayed(onView(withText("LEAVE OFF")))) {
+            // Accept alert for media access
+            clickOn(onView(withText("LEAVE OFF")).inRoot(isDialog()));
+        }
+    }
+
+    public void openSettings() {
+        openActionBarOverflowOrOptionsMenu(getInstrumentation().getTargetContext());
+        clickOn(onView(withText("Post settings")));
+    }
+
+    public void addACategory(String category) {
+        clickOn(onView(withId(R.id.post_categories_container)));
+        clickOn(onView(withText(category)));
+        pressBack();
+    }
+
+    public void addATag(String tag) {
+        clickOn(onView(withId(R.id.post_tags_container)));
+        ViewInteraction tagsField = onView(withId(R.id.tags_edit_text));
+        populateTextField(tagsField, tag);
+        pressBack();
+    }
 
-        return this;
+    public void setFeaturedImage() {
+        clickOn(onView(withId(R.id.post_add_featured_image_button)));
+        clickOn(onView(withId(R.id.icon_wpmedia)));
```

Comment:
The tests are failing on this line for me due to the app not having photos permissions. Here is the screenshot when it fails. The media button isn't present because the photos permission hasn't been given.

![Screenshot_1557834188](https://user-images.githubusercontent.com/1773641/57695375-08dcff00-7646-11e9-9bb4-5602acad4791.png)

You can reset permissions on your device/emulator to reproduce this by running `adb shell pm reset-permissions`.

Reply:
Ahh thanks for catching this!

Reply:
Hmm weird, I thought I removed it in the merge but the rule 

```Java
@Rule
    public GrantPermissionRule mRuntimeImageAccessRule = GrantPermissionRule.grant(permission.READ_EXTERNAL_STORAGE);
```

is there and the permission should be granted, manually walking through the app after the test run also has the permission on but the error message still present 🤔

Reply:
That is strange!

Reply:
In any case, I can add a patch here to click the Allow button if it's present but I'm wondering if I should... Why am I not presented with the option to use my WP media library similarly to when I'm adding an image in the post? I'm thinking I should only need to allow the permission if I want to add a photo from storage... WDYT @jtreanor @rachelmcr?

Reply:
> Why am I not presented with the option to use my WP media library similarly to when I'm adding an image in the post? I'm thinking I should only need to allow the permission if I want to add a photo from storage...

Good catch; seems like an oversight. I'd expect the bottom menu on that screen to always be visible so you could select other sources (e.g. media library) even if you don't grant media permissions.

Reply:
Opened an issue here for it https://github.com/wordpress-mobile/WordPress-Android/issues/9870, I'm thinking to wait on that to be resolved instead of adjusting the test to make it pass.

Reply:
> ```java
> @Rule
> public GrantPermissionRule mRuntimeImageAccessRule = GrantPermissionRule.grant(permission.READ_EXTERNAL_STORAGE);
> ```
> 
> is there and the permission should be granted, manually walking through the app after the test run also has the permission on but the error message still present 🤔

I'm seeing a "permissions denied" message in the app (instead of the "WordPress needs permissions ..." soft ask) when I run the test. The app is checking for `permission.WRITE_EXTERNAL_STORAGE` and switching the test rule to that resolved it for me:

```java
@Rule
public GrantPermissionRule mRuntimeImageAccessRule = GrantPermissionRule.grant(permission.WRITE_EXTERNAL_STORAGE);
```